bgpd/bgp_route.c

```diff
@@ -6930,7 +6930,7 @@ static void route_vty_short_status_out(struct vty *vty,
 		vty_out(vty, " ");
 }
 
-static char *bgp_nexthop_fqdn(struct peer *peer)
+char *bgp_nexthop_fqdn(struct peer *peer)
```

Comment:
show bgp neighbors has some optional key words after neighbor is specified to print advertised-routes, dampened-routes etc. For example, 

```
d8a08f90fafb# show bgp neighbors 
  <cr>      
  A.B.C.D   Neighbor to display information about
     30.1.11.1 30.1.10.1 30.1.13.1 30.1.12.1 30.1.15.1 30.1.14.1 30.1.1.1 30.1.16.2 30.1.3.1 30.1.2.1 30.1.5.1 30.1.4.1 30.1.7.1 
     30.1.6.1 30.1.9.1 30.1.8.1 
  WORD      Neighbor on BGP configured interface
  X:X::X:X  Neighbor to display information about
  json      JavaScript Object Notation
d8a08f90fafb# show bgp neighbors 30.1.11.1 
  <cr>               
  advertised-routes  Display the routes advertised to a BGP neighbor
  dampened-routes    Display the dampened routes received from neighbor
  filtered-routes    Display the filtered routes received from neighbor
  flap-statistics    Display flap statistics of the routes learned from neighbor
  json               JavaScript Object Notation
  prefix-counts      Display detailed prefix count information
  received           Display information received from a BGP neighbor
  received-routes    Display the received routes from neighbor
  routes             Display routes learned from neighbor
```

if user specifies advertised-routes after specifying the host name, how does this work ? does it show all the advertised-routes from all the neighbors from this host ? or these options will be disabled after specifying the host name isntead of neighbor ?

Reply:
For instance:
```
exit1# show ip bgp neighbors spine1-debian-9 routes
BGP table version is 11, local router ID is 172.16.20.254, vrf id 0
Default local pref 100, local AS 200
Status codes:  s suppressed, d damped, h history, * valid, > best, = multipath,
               i internal, r RIB-failure, S Stale, R Removed
Nexthop codes: @NNN nexthop's vrf id, < announce-nh-self
Origin codes:  i - IGP, e - EGP, ? - incomplete

   Network          Next Hop            Metric LocPrf Weight Path
*> 5.5.5.0/24       spine1-debian-9          0             0 100 ?
*> 10.0.0.0/24      spine1-debian-9          0             0 100 ?
*  10.0.2.0/24      spine1-debian-9          0             0 100 ?
*> 10.10.250.0/24   spine1-debian-9          0             0 100 ?
*> 172.16.1.1/32    spine1-debian-9          0             0 100 ?
*  192.168.0.0/24   spine1-debian-9          0             0 100 ?
*> 192.168.1.0/24   spine1-debian-9          0             0 100 ?
*> 192.168.2.0/24   spine1-debian-9          0             0 100 ?
*> 192.168.3.0/24   spine1-debian-9          0             0 100 ?

Displayed  9 routes and 21 total paths
exit1# 
```

Reply:
When a particular host (spine1-debian-9) has more than 1 neighbor, does these commands show the cumulative routes from all neighbors on this host ?

For example, in the above scenario, Do you have multiple neighbors configured from the same host (spine1-debian-9) ? If so the count of above routes (9) is from all the neighbors from the same host ?

Reply:
I didn’t test with multiple neighbors from the same host, but this functionality is out of scope. Basically, it should be handled by peer_lookup_by_hostname() and friends.

Reply:
But answering your question, it doesn't sum the routes from neighbors from the same host:

```
spine1-debian-9# show ip bgp neighbors 192.168.0.1 routes 
BGP table version is 11, local router ID is 172.16.1.1, vrf id 0
Default local pref 100, local AS 100
Status codes:  s suppressed, d damped, h history, * valid, > best, = multipath,
               i internal, r RIB-failure, S Stale, R Removed
Nexthop codes: @NNN nexthop's vrf id, < announce-nh-self
Origin codes:  i - IGP, e - EGP, ? - incomplete

   Network          Next Hop            Metric LocPrf Weight Path
*  10.0.2.0/24      exit1-debian         0             0 200 ?
*> 172.16.20.254/32 exit1-debian         0             0 200 ?

Displayed  2 routes and 12 total paths

spine1-debian-9# show ip bgp neighbors 2a02:4780:face::2 routes 
BGP table version is 11, local router ID is 172.16.1.1, vrf id 0
Default local pref 100, local AS 100
Status codes:  s suppressed, d damped, h history, * valid, > best, = multipath,
               i internal, r RIB-failure, S Stale, R Removed
Nexthop codes: @NNN nexthop's vrf id, < announce-nh-self
Origin codes:  i - IGP, e - EGP, ? - incomplete

   Network          Next Hop            Metric LocPrf Weight Path
*  192.168.0.0/24   exit1-debian         0             0 200 ?

Displayed  1 routes and 12 total paths

spine1-debian-9# show ip bgp neighbors exit1-debian routes 
BGP table version is 11, local router ID is 172.16.1.1, vrf id 0
Default local pref 100, local AS 100
Status codes:  s suppressed, d damped, h history, * valid, > best, = multipath,
               i internal, r RIB-failure, S Stale, R Removed
Nexthop codes: @NNN nexthop's vrf id, < announce-nh-self
Origin codes:  i - IGP, e - EGP, ? - incomplete

   Network          Next Hop            Metric LocPrf Weight Path
*  10.0.2.0/24      exit1-debian         0             0 200 ?
*> 172.16.20.254/32 exit1-debian         0             0 200 ?

Displayed  2 routes and 12 total paths
```

Reply:
I think this can be fixed with further PRs, what do you think?

Reply:
I think it's okay to open a new PR for this issue rather than including it here...